nebula_decoders/tests/point_filters/test_downsample_mask.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nebula_decoders/nebula_decoders_common/angles.hpp"
+#include "nebula_decoders/nebula_decoders_common/point_filters/downsample_mask.hpp"
+
+#include <nebula_common/loggers/console_logger.hpp>
+#include <nebula_common/nebula_common.hpp>
+#include <nebula_common/point_types.hpp>
+#include <png++/gray_pixel.hpp>
+#include <png++/image.hpp>
+
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <filesystem>
+#include <limits>
+#include <memory>
+#include <string>
+
```

Comment:
Pre-commit is also requesting 
`#include <vector>`

Reply:
Addressed in a85ea4b